sources/osgShadow/ShadowMapAtlas.js

```diff
@@ -0,0 +1,600 @@
+'use strict';
+var Notify = require( 'osg/notify' );
+var StateAttribute = require( 'osg/StateAttribute' );
+var Texture = require( 'osg/Texture' );
+var Uniform = require( 'osg/Uniform' );
+var MACROUTILS = require( 'osg/Utils' );
+var Scissor = require( 'osg/Scissor' );
+var vec4 = require( 'osg/glMatrix' ).vec4;
+var ShadowTechnique = require( 'osgShadow/ShadowTechnique' );
+var ShadowTextureAtlas = require( 'osgShadow/ShadowTextureAtlas' );
+var ShadowMap = require( 'osgShadow/ShadowMap' );
+
+/**
+ *  ShadowMapAtlas provides an implementation of shadow textures.
+ * here, one shadow
+ *  @class ShadowMapAtlas
+ */
+var ShadowMapAtlas = function ( settings ) {
+
+    this._lights = [];
+    this._shadowMaps = [];
+    this._viewportDimension = [];
+
+    ShadowTechnique.apply( this, arguments );
+
+    this._texture = new ShadowTextureAtlas();
+    this._textureUnitBase = 4;
+    this._textureUnit = this._textureUnitBase;
+
+    // see shadowSettings.js header for param explanations
+    this._textureMagFilter = undefined;
+    this._textureMinFilter = undefined;
+
+
+    this._textureSize = 1024;
+    this._shadowMapSize = 256;
+
+    this._receivingStateset = undefined;
+
+    this._shaderProcessor = undefined;
+
+    if ( settings ) {
+
+        this.setShadowSettings( settings );
+        if ( settings.atlasSize ) this._textureSize = settings.atlasSize;
+        if ( settings.textureSize ) this._shadowMapSize = settings.textureSize;
+
+    }
+
+    this._texelSizeUniform = Uniform.createFloat1( 1.0 / this._textureSize, 'texelSize' );
+
+    var unifRenderSize = Uniform.createFloat2( 'RenderSize' );
+    this._renderSize = unifRenderSize.getInternalArray();
+    this._renderSize[ 0 ] = this._renderSize[ 1 ] = this._textureSize;
+
+    this._numShadowWidth = this._textureSize / this._shadowMapSize;
+    this._numShadowHeight = this._textureSize / this._shadowMapSize;
+
+};
+
+
+/** @lends ShadowMapAtlas.prototype */
+ShadowMapAtlas.prototype = MACROUTILS.objectLibraryClass( MACROUTILS.objectInherit( ShadowTechnique.prototype, {
+
+    getDepthRange: function ( numShadow ) {
+        return this._shadowMaps[ numShadow ].getDepthRange();
+    },
+
+    getTexture: function () {
+        return this._texture;
+    },
+
+    isDirty: function ( numShadow ) {
+        if ( numShadow !== undefined ) {
+            return this._shadowMaps[ numShadow ].isDirty();
```

Comment:
it does not seem to be numShadow but instead an index. It should not be named this way, it can be confuse with the number used for light.